File: ports/rust/src/core/containers/bcsvector.rs
// BcsVector is a generic dynamically resizing array, mapped to Rust's Vec
pub struct BcsVector<T> {
    items: Vec<T>,
}

impl<T> BcsVector<T> {
    pub fn new() -> Self {
        BcsVector { items: Vec::new() }
    }

    pub fn append(&mut self, item: T) {
        self.items.push(item);
    }

    pub fn at(&self, index: usize) -> Option<&T> {
        self.items.get(index)
    }

    pub fn size(&self) -> usize {
        self.items.len()
    }

    pub fn clear(&mut self) {
        self.items.clear();
    }
}

File: ports/go/src/core/containers/bcsvector.go
package containers

// BcsVector is a generic dynamically resizing array, mapped to Go slices.
type BcsVector[T any] struct {
	items []T
}

// NewBcsVector creates a new empty vector.
func NewBcsVector[T any]() *BcsVector[T] {
	return &BcsVector[T]{
		items: make([]T, 0),
	}
}

// Append adds an item to the end of the vector.
func (v *BcsVector[T]) Append(item T) {
	v.items = append(v.items, item)
}

// At returns the item at the specified index.
func (v *BcsVector[T]) At(index int) T {
	return v.items[index]
}

// Size returns the number of items in the vector.
func (v *BcsVector[T]) Size() int {
	return len(v.items)
}

// Clear removes all elements from the vector.
func (v *BcsVector[T]) Clear() {
	v.items = make([]T, 0)
}
